apps/api_web/test/api_web/controllers/route_pattern_controller_test.exs

```diff
@@ -253,6 +254,34 @@ defmodule ApiWeb.RoutePatternControllerTest do
       assert [] == json_response(conn, 200)["data"]
     end
 
+    test "can filter by is_canonical", %{conn: conn} do
+      State.RoutePattern.new_state([
+        %RoutePattern{id: "rp1", route_id: "route1", direction_id: 0, is_canonical: 1},
+        %RoutePattern{id: "rp2", route_id: "route1", direction_id: 1, is_canonical: 1},
+        %RoutePattern{id: "rp3", route_id: "route2", direction_id: 0, is_canonical: 2},
+        %RoutePattern{id: "rp4", route_id: "route2", direction_id: 1, is_canonical: 2}
+      ])
+
+      conn =
+        get(
+          conn,
+          route_pattern_path(conn, :index, %{
+            "filter" => %{"is_canonical" => true}
+          })
+        )
+
+      assert [
+               %{
+                 "id" => "rp1",
+                 "attributes" => %{"direction_id" => 0, "is_canonical" => true}
+               },
+               %{
+                 "id" => "rp2",
+                 "attributes" => %{"direction_id" => 1, "is_canonical" => true}
+               }
+             ] = json_response(conn, 200)["data"]
+    end
```

Comment:
suggestion: We should probably test route pattern filtering when `is_canonical = 0`.